config/kafka/kafka.yaml

```diff
@@ -69,13 +86,15 @@ metadata:
     component: kafka-broker
   name: projectriff-kafka
 spec:
-  replicas: 1
+  replicas: 1 # The headless service/hostname config only supports 1 replica!
   template:
     metadata:
       labels:
         app: kafka
         component: kafka-broker
     spec:
+      hostname: broker-1
```

Comment:
I don't understand this. Why force a hostname?

Reply:
See my comment below about having only one replica. This should make this unnecessary.

Also, how would this play out if Envoy was sitting in the middle?